src/closure/createClosure.ts

```diff
@@ -1038,18 +1088,34 @@ async function getOrCreateEntryAsync(
     if (normalizedModuleName) {
       await captureModuleAsync(normalizedModuleName);
     } else if (obj instanceof Function) {
-      // Serialize functions recursively, and store them in a closure property.
-      entry.function = await analyzeFunctionInfoAsync(
-        obj,
-        context,
-        serialize,
-        logInfo
-      );
+      // if not invoked, let the property be treated like an object
+      let serializeAll = true;
+
+      if (
+        typeof isInvoked === "boolean" &&
+        !isInvoked &&
+        capturedObjectProperties &&
+        capturedObjectProperties.length > 0
+      ) {
+        entry.object = entry.object || { env: new Map() };
```

Comment:
Do you mean to use `??` instead of `||`?

Reply:
Why `new Map()`?

Reply:
I copies this from elsewhere in the code, lol, but it appears that object needs to be a map or else things start to fail.